artifacthub-repo.yml

```diff
@@ -0,0 +1,7 @@
+# Artifact Hub repository metadata file
+# https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-repo.yml
+
+repositoryID: a4a4aff7-52f0-43f9-93a5-0749f53bc898
```

Comment:
> This ID is proper to us?

Yes, it is the ID that the [Artifact Hub control panel](https://artifacthub.io/control-panel) gave when I registered https://github.com/inspektor-gadget/inspektor-gadget/.

![image](https://github.com/inspektor-gadget/inspektor-gadget/assets/27575/21bd3637-573a-420a-abf8-47ad5c40da48)

> What happens if I fork?

Nothing will happen with the fork because the config in the [Artifact Hub control panel](https://artifacthub.io/control-panel) only looks at https://github.com/inspektor-gadget/inspektor-gadget/.

The Artifact Hub bot is not looking at all repositories on GitHub but only the ones that someone registered in the control panel.

If a fork wants to register their fork in Artifact Hub, they will have to register it themselves with their own GitHub account (and not with the inspektor-gadget org). Then Artifact Hub will give a different repositoryID and the fork will have to update `artifacthub-repo.yml` in the fork to that `repositoryID`.

> Will I be able to push artifacts to the official hub?

Artifact Hub is not hosting artifacts but just references them. The OCI images stay on ghcr.io, they are not duplicated on Artifact Hub.

If you have write access to https://github.com/inspektor-gadget/inspektor-gadget/, you can update the `artifacthub-repo.yml` in git and the Artifact Hub bot will update its database at the next scrape.

If your GitHub user belongs to inspektor-gadget org, you can authenticate on [Artifact Hub control panel](https://artifacthub.io/control-panel) and change the configuration, so it scrapes a different github repo, or a different branch. At the moment, it is configured to use the `alban_hub` branch. I will update the configuration to use `main` after this PR is merged.

> Is it possible to make this file configurable? i.e. something like:
> repositoryID: ${{ secrets.artifacthubID }}

It is not possible. Artifact Hub database is not updated by a GitHub action in our repository. Instead, the Artifact Hub bot is configured to pull the configured repositories every 30 minutes. So we need at least a branch (e.g. `main`) to have a the correct `repositoryID` when it is scraped by the bot.